nemo/collections/asr/parts/submodules/fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,440 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nemo.core.utils.cuda_python_utils import check_cuda_python_cuda_graphs_conditional_nodes_supported
+
+check_cuda_python_cuda_graphs_conditional_nodes_supported()
+
+import contextlib
+import ctypes
+import time
+from dataclasses import dataclass, field
+from itertools import product
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from cuda import cuda, cudart, nvrtc
```

Comment:
It is not guaranteed that pytorch-cuda will be installed by user, please add a static try catch check 

try:
  from cuda import xyz
  HAVE_CUDA_PYTHON=True
except (Import exception, module not found write):
  HAVE_CUDA_PYTHON=False

Reply:
So the static try catch at the top is the global flag at python instantiation time whether the library could be imported or not. It is safe to import this file when if the bool is set to false and the library is not available. 

However a user might try to run cuda graph op without installing it, then at init of the class, which is instantiated dynamically at runtime, we can check the function to assert that the library is actually installed correctly, otherwise raise an appropriate error

Reply:
So yes, we'd need both. Static check outside unit inside file, dynamic check with function inside init.